### PR TITLE
75 fix highlight all behaviour

### PR DESCRIPTION
**Pull request by andreascnits**
[fixed button interaction with highlighting and highlightAll. Also moved CollapsibleItemElement into CollapsibleItemType for easier interaciton between the two.

### CODE REVIEW

File: src/messageObjects/message.tsx [hunk 1]
```diff
@@ -12,8 +12,24 @@ export class HighlightMessage implements Message {
...
+    }
+    action: string = "unhighlightAllAndHighlightSingle";
+}
+
 export class HighlightAllMessage implements Message {
-    constructor(public type: ElementType, public isChecked: boolean){
+    constructor(public type: ElementType, public isChecked: boolean) {
         this.type = type;
         this.isChecked = isChecked;
     }
```
Comment (theaueland) [inline src/messageObjects/message.tsx:35]: Do you need to create a new class for every message?

**Review by theaueland — CHANGES_REQUESTED** [review]
I added a few comments you should look at. Other than that it looks

File: src/messageObjects/message.tsx (hunk 1 above)
Comment (andreascnits) [reply src/messageObjects/message.tsx:35]: Each new class is made to handle different payload. To make sure that each case only checks for the right elements.

An alternative would be to just have a field payload, but then that payload would have to take in alot of different types.
Might be other ways to do it, but i thought this was an easy and safe way to do it

File: src/contentScript.tsx [hunk 2]
```diff
@@ -22,52 +22,77 @@ document.head.append(styleElement);
...
+        }
+      });
+      console.log("Antall knapper uten alt tekst:", noAltTextCount);
+      chrome.storage.local.set({ noAltTextCount: noAltTextCount } as { noAltTextCount: number },
+        () => {
+          chrome.runtime.sendMessage({ action: "noAltTextCountUpdated", count: noAltTextCount })
+        })
+      sendResponse({ message: "Button alternative text checked" });
+
+    } else */
```
[the remark this answers is not in the document]
Comment (andreascnits) [reply src/contentScript.tsx]: removed auduns old code and saved locally

**Review by theaueland — APPROVED** [review]

**Pull request merged**